File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    int n;
    do
    {
        n = get_int("Pyramid number =");
    }
    while (n < 0 || n > 23);

// Here we start printing the columns
    for (int i = 0; i < n; i++)
    {
        // Here we start printing the space rows
        for (int j = n; j > i + 1; j--)
        {
            printf(" ");
        }
        // Here we stare printing the # rows
        for (int j = i; j > 0; j--)
        {
            printf("#");
        }
        // Finally the 2 # at the end
        printf("#  #");

        for (int j = i; j > 0; j--)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: mario.py

from cs50 import get_int

def main():
    while True:
        n = get_int("Pyramid number =")
        if n > 0 or n <= 23:
            break
    for i in range (n):
        for j in range (n - 1 - i):
            print(" ", end = "")
        for j in range (i):
            print("#", end = "")
        print ("# #", end = "")
        for j in range (i):
            print ("#", end = "")
        print()

if __name__ == "__main__":
    main()
